pkg/network/network.go

```diff
@@ -17,39 +17,53 @@ func (s *SkupperStatus) GetServiceSitesMap() map[string][]SiteStatusInfo {
 	for _, site := range s.NetworkStatus.SiteStatus {
 
 		if len(site.RouterStatus) > 0 {
-			for _, listener := range site.RouterStatus[0].Listeners {
-				if mapServiceSites[listener.Name] != nil {
-					serviceSites := mapServiceSites[listener.Name]
-
-					serviceSites = append(serviceSites, site)
-					mapServiceSites[listener.Name] = serviceSites
-				} else {
-					mapServiceSites[listener.Name] = []SiteStatusInfo{site}
+			for _, router := range site.RouterStatus {
+				for _, listener := range router.Listeners {
+					if len(mapServiceSites[listener.Address]) == 0 {
+						mapServiceSites[listener.Address] = []SiteStatusInfo{site}
+					} else if mapServiceSites[listener.Address] != nil && !sliceContainsSite(mapServiceSites[listener.Address], site) {
+						serviceSites := mapServiceSites[listener.Address]
```

Comment:
iiuc suggest:
mapServiceSites[listener.Address] = append(mapServiceSites[listener.Address], site)

Reply:
It is indeed more readable now. Thanks! Updated in f2f23f1b0cde2c96ab96ffc027480a1c05f1899b